server/app/services/rpc_server.rb

```diff
@@ -13,30 +22,76 @@ def initialize(code, message, backtrace = nil)
     end
   end
 
-  ##
+  def initialize
+    @handlers = {}
+  end
+
+  # @param [Faye::Websocket] ws_client
   # @param [String] grid_id
   # @param [Array] message msgpack-rpc request array
   # @return [Array]
-  def self.handle_request(grid_id, message)
+  def handle_request(ws_client, grid_id, message)
     msg_id = message[1]
-    grid = Grid.find(grid_id)
-    return [1, msg_id, {error: 'grid not found'}, nil] if grid.nil?
-
     handler = message[2].split('/')[1]
     method = message[2].split('/')[2]
-    if klass = HANDLERS[handler]
+    if actor = handling_actor(grid_id, handler)
       begin
-        result = klass.new(grid).send(method, *message[3])
-        return [1, msg_id, nil, result]
+        result = actor.send(method, *message[3])
+        send_message(ws_client, [1, msg_id, nil, result])
+      rescue Celluloid::DeadActorError
+        error "actor for handler #{handler} is dead, removing it from cache"
+        @handlers[grid_id].delete(handler)
       rescue RpcServer::Error => exc
```

Comment:
The handlers can't really raise `RpcServer::Error` anymore if they are separate actors.